src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -560,6 +565,27 @@ ErrorIfNotSupportedForeignConstraint(Relation relation, char distributionMethod,
 			continue;
 		}
 
+		referencedTableId = constraintForm->confrelid;
+		selfReferencingTable = relation->rd_id == referencedTableId;
+
+		/*
+		 * We do not support foreign keys for reference tables. Here we skip the second
+		 * part of check if the table is a self referencing table because;
+		 * - PartitionMethod only works for distributed tables and this table is not
+		 * distributed yet.
+		 * - Since referencing and referenced tables are same, it is OK to not checking
+		 * distribution method twice.
+		 */
+		if (distributionMethod == DISTRIBUTE_BY_NONE ||
+			(!selfReferencingTable &&
```

Comment:
What happens if we remove `! selfReferencingTable ` table from the if statement? Is this just a performance improvement? If so, could we remove that?